server/src/services/logging/SubRequestMapper.ts

```diff
@@ -0,0 +1,35 @@
+import { SubRequestData } from "./RequestData.ts";
+import { uuid } from " https://deno.land/x/uuid/mod.ts";
+import { FetchData } from "../validation/api/Client.ts";
+
+export class SubRequestMapper {
+  public map = (data: FetchData, headers: Record<string, string>, response: Response, date: Date): void => {
+    let requestOptions: any = {
+      method: data.method,
+      headers: headers,
+    };
+    if (data.method !== "GET") {
+      requestOptions = {
+        ...requestOptions,
+        body: JSON.stringify(data.body),
+      }
+    }
+    const request = new Request(data.url, requestOptions);
+    data.context.subrequests.push(new SubRequestData({
```

Comment:
you don't need the `SubRequestData`, you can build the data type needed for logger from req/res and context